libvast/src/segment_store.cpp

```diff
@@ -86,6 +86,94 @@ caf::error segment_store::flush() {
   return save(sys_, meta_path(), segments_);
 }
 
+caf::expected<segment_ptr> segment_store::load_segment(uuid id) const {
+  segment_ptr seg_ptr = nullptr;
+  auto fname = segment_path() / to_string(id);
+  if (auto err = load(sys_, fname, seg_ptr)) {
+    VAST_ERROR(this, "unable to load segment:", sys_.render(err));
+    return err;
+  }
+  return seg_ptr;
+}
+
+std::unique_ptr<store::lookup> segment_store::extract(const ids& xs) const {
+
+  class lookup : public store::lookup {
+  public:
+    using uuid_iterator = std::vector<uuid>::iterator;
+
+    lookup(const segment_store& store, ids xs, std::vector<uuid>&& candidates)
+      : store_{store}, xs_{std::move(xs)}, candidates_{candidates} {
+      // nop
+    }
+
+    caf::expected<table_slice_ptr> next() override {
+      // update the slice buffer if ...
+      while (!buffer                // ... the previous lookup failed.
+          || it_ == buffer->end())  // ... the buffer has been consumed.
```

Comment:
I appreciate the granular commenting, but I think it remains readable if you put the comment in one chunk at the top. I find the broken `while` statement actually less readable.